### PR TITLE
vine: add dask-colors option and fontsize option

### PR DESCRIPTION
**Pull request by JinZhou5042**
## Proposed changes

This PR adds two options into `vine_plot_txn_log.py`, one is `--dask-colors` that enables us to produce a figure in which color palette aligns with Dask's, one is `--fontsize` by which we can set the label size of the figure legend.

## Post-change actions

Put an 'x' in the boxes that describe post-change actions that you have done.
The more 'x' ticked, the faster your changes are accepted by maintainers.

- [x] `make test`       Run local tests prior to pushing.
- [ ] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [ ] `make lint`       Run lint on source code prior to pushing.
- [ ] Manual Update     Did you update the manual to reflect your changes, if appropriate? This action should be done after your changes are approved but not merged.
- [x] Type Labels       Select github labels for the type of this change: bug, enhancement, etc.
- [x] Product Labels    Select github labels for the product affected: TaskVine, Makeflow, etc.
- [ ] PR RTM            Mark your PR as ready to merge.

## Additional comments
This section is dedicated to changes that are ambitious or complex and require substantial discussions. Feel free to start the ball rolling.

### CODE REVIEW

**Comment by JinZhou5042** [comment]
`make format` and `make lint` both failed, should I pay attention to them?

Screenshot from 2023-10-16 11-27-24

File: taskvine/src/tools/vine_plot_txn_log [hunk 1]
```diff
@@ -435,6 +435,12 @@ class TxnPlot:
                 "outputs to manager": 'C5',
                 }
 
+        # dask legend
+        self.legend_dask = {
```
Comment (btovar) [inline taskvine/src/tools/vine_plot_txn_log:439]: I suggest:
Rename self.legend to self.legend_cb

```python

self.legend_dask = dict(self.legend_cb)
self.legend_dask.update(
{
 "task executing": '#7fc .... etc.
})
self.legend = self.legend_cb
```

Then, if `--dask-colors` is specified, simply assign `self.legend = self.legend_dask`. With that I don't think you need to change any of the functions `bh`,  `set_legend`, or  `plot`, nor create `plot_dask`.

**Review by btovar — CHANGES_REQUESTED** [review]

**Comment by btovar** [comment]
Yes, `make format` and `make lint` should not have any errors.

**Comment by btovar** [comment]
Also, the commit history should be fixed as it includes commits that are not part of this pr.

**Comment by dthain** [comment]
Your `make format` errors suggest you don't have the formatting tools installed.  (This has changed recently, check `environment.yml`

File: taskvine/src/tools/vine_plot_txn_log (hunk 1 above)
Comment (btovar) [reply taskvine/src/tools/vine_plot_txn_log:439]: That matplotlib palette is supposed to be friendly to **c**olor**b**lind persons. (e.g. https://viscid-hub.github.io/Viscid-docs/docs/dev/styles/tableau-colorblind10.html)

Comment (JinZhou5042) [reply taskvine/src/tools/vine_plot_txn_log:439]: Great! Thanks!

Comment (JinZhou5042) [reply taskvine/src/tools/vine_plot_txn_log:439]: This may have conflicts with my current idea... I changed `bh` because the edgecolor needs to align with Dask's, and I changed `plot_dask`, because in Dask's figure there are only two kinds of bars and we don't need to plot the worker lifetime, and `set_legend` for the same reason.

image

**Comment by JinZhou5042** [comment]
> I think that if a legend keyword is not in the legend dictionary, then it is not plotted. So you can remove the keys from legend_dask that you don't need (e.g. with `del`).

Yes! By doing so, now I can make the functions' execution logic much clearer! But I think I still need to reserve those changes in `set_legend`,  `plot` and `plot_dask`. Do you have some suggestions?

File: taskvine/src/tools/vine_plot_txn_log [hunk 2]
```diff
@@ -565,7 +587,11 @@ class TxnPlot:
     def set_legend(self, fig):
         if self.opts.legend != 'none':
             handles = [mpl.patches.Patch(color=c, label=l) for l,c in self.legend.items()]
-            fig.legend(loc=self.opts.legend, handles=handles)
+
+            fontsize = 'medium'
+            if args.fontsize:
+                fontsize = args.fontsize
+            fig.legend(loc=self.opts.legend, handles=handles, fontsize=fontsize)
```
Comment (btovar) [inline taskvine/src/tools/vine_plot_txn_log:594]: Use fontsize=args.fontsize.   You don't need the `if` nor the variable `fontsize`, as medium is already the default.

File: taskvine/src/tools/vine_plot_txn_log [hunk 3]
```diff
@@ -687,31 +713,55 @@ class TxnPlotWorkers(TxnPlot):
...
-
-        base_slot=0
-        for wid in by_min_connects:
-            worker = gsd.get(wid, None)
-            worker_times = ws[ws["worker_id"] == wid]
-            worker_xfers = xs[xs["worker_id"] == wid]
-
-            if worker is not None:
-                base_slot = self.plot_by_worker(fig, worker, worker_times, worker_xfers, base_slot)
+        tasks = manager.tasks
```
Comment (btovar) [inline taskvine/src/tools/vine_plot_txn_log:740]: Making the names clearer is fine, however do not include these unrelated changes in this pr. Otherwise it makes it harder to evaluate the main objective of the pr, and may introduce unrelated bugs.

**Review by btovar — CHANGES_REQUESTED** [review]

File: taskvine/src/tools/vine_plot_txn_log (hunk 3 above)
Comment (JinZhou5042) [reply taskvine/src/tools/vine_plot_txn_log:740]: Sure! And could you give me a hint about how to remove those unrelated commits?
